src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -341,6 +341,33 @@
       result
       (recur db-spec))))
 
+(defn garbage-collector
+  [db clean-lock config]
+  (let [seconds-pos? (comp pos? to-seconds)
+        what (filter identity
+                     [(when-let [node-ttl (:node-ttl config)]
+                        (when (seconds-pos? node-ttl)
+                          "expire_nodes"))
+                      (when-let [node-purge-ttl (:node-purge-ttl config)]
+                        (when (seconds-pos? node-purge-ttl)
+                          (let [limit (:node-purge-gc-batch-limit config)]
+                            (if (zero? limit)
+                              "purge_nodes"
+                              ["purge_nodes" {:batch_limit limit}]))))
+                      (when-let [report-ttl (:report-ttl config)]
+                        (when (seconds-pos? report-ttl)
+                          "purge_reports"))
+                      "other"])]
+    (fn []
```

Comment:
can we make this a regular function? And use `#()` notation to create the thunk for at-at